vllm/model_executor/sampling_metadata.py

```diff
@@ -466,22 +465,30 @@ def from_lists(cls, temperatures: List[float], top_ps: List[float],
         do_penalties = prompt_tokens or output_tokens
 
         if do_penalties:
-            prompt_max_len = max([len(tokens) for tokens in prompt_tokens],
+            prompt_max_len = max((len(tokens) for tokens in prompt_tokens),
                                  default=0)
-            prompt_padded_tokens = np.full(
-                (len(prompt_tokens), prompt_max_len),
+            prompt_t = make_tensor_with_pad(
+                prompt_tokens,
+                prompt_max_len,
                 vocab_size,
-                dtype=np.int64)
-            for i, tokens in enumerate(prompt_tokens):
-                prompt_padded_tokens[i, :len(tokens)] = tokens
-            output_max_len = max([len(tokens) for tokens in output_tokens],
+                device="cpu",
+                dtype=torch.int64,
+                pin_memory=pin_memory,
```

Comment:
Should we just initialize the tensor on the target device directly?

Reply:
Better to use non-blocking transfer, because even we directly allocate on GPU, it still need to initialize on CPU and then transfer to GPU I think.

Reply:
Alright, I'll keep the code as is then.